### PR TITLE
Changed Lutris asset folder to .cache

### PR DESCRIPTION
**Pull request by Blood-Pirouette**
Fixed Issue #1073. I added a function to look in the cache folder for lutris assets and to use them. Tested it and it seems to be working now

Disclaimer: This is my first PR and my knowledge on CPP is limited so please merge only if appropriate :+1: Thanks :)

### CODE REVIEW

**Comment by mmatyas** [comment]
Thanks for the fix! It might be that something changed in Lutris, so I'd like to take a look in the source code first sometime.

**Comment by mmatyas** [comment]
Really sorry this took so long, I've commented in the issue report.

File: src/backend/providers/lutris/LutrisProvider.cpp [hunk 1]
```diff
@@ -51,6 +51,17 @@ QString find_datadir()
...
+{
+    const QStringList asset_roots = QStandardPaths::standardLocations(QStandardPaths::GenericCacheLocation);
+    for (const QString& asset_path : asset_roots) {
+        QString check_assetdir = asset_path + QLatin1String("/lutris/banners/");  
+        QString assetdir = asset_path + QLatin1String("/lutris/");
+        if (QFileInfo::exists(check_assetdir)) //Checks if the directory ~/.cache/lutris/banners exists
+            return assetdir;
+    }
+    return find_datadir(); //Returns the datadir instead when the above loop did not find assets under cache
+}
```
Comment (mmatyas) [inline src/backend/providers/lutris/LutrisProvider.cpp:64]: There's no need for two different paths in the `for` loop. If we call the loop variable as `root_path`, like in `find_datadir`, then we only have to check for "assetdir = root path + /lutris/banners/".

If there are no hits, we should return an empty string. We can then handle that case in the calling function.

Comment (Blood-Pirouette) [reply src/backend/providers/lutris/LutrisProvider.cpp:65]: Added paths `assetdir` and `banner_subdir`. The code will check for the existence of `banner_subdir` then return assetdir. If there are not hits, an empty string is returned.

File: src/backend/providers/lutris/LutrisProvider.cpp [hunk 2]
```diff
@@ -107,6 +118,13 @@ Provider& LutrisProvider::run(SearchContext& sctx)
     }
     Log::info(display_name(), LOGMSG("Found data directory: `%1`").arg(datadir));
 
+    const QString assetdir = find_assetdir();
+    if (assetdir.isEmpty()) {
+        Log::info(display_name(), LOGMSG("No lutris assets found"));
+        return *this;
+    }
+
+
```
[the remark this answers is not in the document]
Comment (Blood-Pirouette) [reply src/backend/providers/lutris/LutrisProvider.cpp:127]: Removed `Log` and `return *this` and assigned assetdir to datadir when `find_assetdir()` returns an empty string.

**Comment by mmatyas** [comment]
Ok, looks good! Now a little cleanup for the end:

- There are 4 separate commits on your branch: Ideally work-in-progress commits are joined into one commit once everything is done. Would you like to squash your commits into one? I can do this too for you, if you don't feel that confident in Git yet.
- Code style: There are 2 empty lines at lines 126–127, could you remove one of them? In most part of Pegasus, only one empty line is used inside functions.

**Comment by Blood-Pirouette** [comment]
@mmatyas I squashed the commits into one and removed the extra line.

**Review by mmatyas — APPROVED** [review]
Nice! :+1:

**Pull request merged**